internal/pkg/manifest/workload_test.go

```diff
@@ -679,7 +791,51 @@ network:
 					Placement: PlacementArgOrString{
 						PlacementString: placementStringP(PublicSubnetPlacement),
 					},
-					SecurityGroups: []string{"sg-1234", "sg-4567"},
+					SecurityGroups: SecurityGroupsIDsOrConfig{
+						IDs:            []string{"sg-1234", "sg-4567"},
+						AdvancedConfig: SecurityGroupsConfig{},
+					},
+				},
+			},
+		},
+		"unmarshal is successful for security groups specified in config": {
```

Comment:
love these tests!! super nice to visualize how manifest is going to be unmarshalled!